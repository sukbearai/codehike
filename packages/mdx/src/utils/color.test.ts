import { describe, it, expect, test } from "vitest"

import { hexToObject, objectToHex } from "./color"

describe("hexToObject", () => {
  it("works with three chars", () => {
    expect(hexToObject("#f8f")).toMatchSnapshot()
  })
  it("works with four chars", () => {
    expect(hexToObject("#f8f8")).toMatchSnapshot()
  })
  it("works with six chars", () => {
    expect(hexToObject("#ff88ff")).toMatchSnapshot()
  })
  it("works with eight chars", () => {
    expect(hexToObject("#ff88ff88")).toMatchSnapshot()
  })
  it("works with undefined", () => {
    expect(hexToObject(undefined)).toMatchSnapshot()
  })
  it("throws with wrong hex", () => {
    expect(() => hexToObject("foo")).toThrow()
    expect(() => hexToObject("#2")).toThrow()
    expect(() => hexToObject("rgb(1,2,3)")).toThrow()
  })
})

describe("objectToHex", () => {
  it("works", () => {
    expect(
      objectToHex({ r: 255, g: 136, b: 255, a: 0.5 })
    ).toBe("#ff88ff80")
  })
  it("works with undefined", () => {
    expect(objectToHex(undefined)).toBe(undefined)
  })
})
