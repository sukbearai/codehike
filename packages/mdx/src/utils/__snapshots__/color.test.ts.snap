// Vitest Snapshot v1

exports[`hexToObject > works with eight chars 1`] = `
{
  "a": 0.5333333333333333,
  "b": 255,
  "g": 136,
  "r": 255,
}
`;

exports[`hexToObject > works with four chars 1`] = `
{
  "a": 0.5333333333333333,
  "b": 255,
  "g": 136,
  "r": 255,
}
`;

exports[`hexToObject > works with six chars 1`] = `
{
  "a": 1,
  "b": 255,
  "g": 136,
  "r": 255,
}
`;

exports[`hexToObject > works with three chars 1`] = `
{
  "a": 1,
  "b": 255,
  "g": 136,
  "r": 255,
}
`;

exports[`hexToObject > works with undefined 1`] = `undefined`;
